appeals/web/src/server/appeals/appeal-details/service-user/service-user.controller.js

```diff
@@ -60,10 +65,15 @@ export const postChangeServiceUser = async (request, response) => {
 	if (request.errors) {
 		return renderChangeServiceUser(request, response);
 	}
-
+	const backToMenuUrl = request.originalUrl.split('/').slice(0, -3).join('/');
 	try {
 		// @ts-ignore
-		const serviceUserId = appealDetails[userType]?.id;
+		const serviceUserId = appealDetails[userType]?.serviceUserId;
```

Comment:
Is there any way to avoid this `ts-ignore`?

Reply:
Unfortunately not with listing every key. But even then I cannot control what the user enters. There ends up being a ts-ignore at some point. At least here you don't list the valid userTypes which means you can add more if you want to. The userType is hardcoded in the mapper so agent or appellant.